lib/hal/led.c

```diff
@@ -0,0 +1,27 @@
+#include <hal/led.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <assert.h>
+
+#define SMC_REG_LEDMODE 0x07
+#define SMC_REG_LEDSEQ  0x08
+
+void XSetAutoLED(void) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x00);
+}
+
+void XSetManualLED(unsigned int t1, unsigned int t2, unsigned int t3, unsigned int t4) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x01);
+
+  assert((t1 & ~0x11) == 0x00);
+  assert((t2 & ~0x11) == 0x00);
+  assert((t3 & ~0x11) == 0x00);
+  assert((t4 & ~0x11) == 0x00);
```

Comment:
nit: I'd use `== 0` instead of `0x00` as the intention is specifically to check for 0 outside of the 0x11 pattern, so specifically `0x000000...` (of unknown size, hence: `0`)